### PR TITLE
Updates/jan 17 sumedh

### PR DESCRIPTION
**Pull request by sumedh-mohod**
counselor session and session feedback done

### CODE REVIEW

File: app/Http/Controllers/CounselorSessionController.php [hunk 1]
```diff
@@ -15,13 +15,14 @@ public function store(Request $request, $id)
...
             'meeting_name' => 'required',
+            'purpose_session' => 'required',
         ]);
         CounselorSession::create([
             'user_id' => $id,
             'session_date' => $request->get('session_date'),
             'session_time' => $request->get('session_time'),
             'meeting_name' => $request->get('meeting_name'),
-            'meeting_option' => $request->get('meeting_option'),
+            'purpose_session' => $request->get('purpose_session'),
```
Comment (ithqindia) [inline app/Http/Controllers/CounselorSessionController.php:25]: purpose_session -> session_purpose

File: app/Http/Controllers/CounselorSessionController.php [hunk 2]
```diff
@@ -56,12 +57,14 @@ public function update(Request $request, $id)
             'session_date' => 'required',
             'session_time' => 'required',
             'meeting_name' => 'required',
+            'purpose_session' => 'required',
```
Comment (ithqindia) [inline app/Http/Controllers/CounselorSessionController.php:60]: purpose_session -> session_purpose

File: app/Http/Controllers/CounselorSessionController.php [hunk 3]
```diff
@@ -56,12 +57,14 @@ public function update(Request $request, $id)
...
         ]);
         $counselorSessionInformation->$id;
         $counselorSessionInformation->session_date = $request->get('session_date');
         $counselorSessionInformation->session_time = $request->get('session_time');
         $counselorSessionInformation->meeting_name = $request->get('meeting_name');
         $counselorSessionInformation->meeting_option = $request->get('meeting_option');
+        $counselorSessionInformation->purpose_session = $request->get('purpose_session');
```
Comment (ithqindia) [inline app/Http/Controllers/CounselorSessionController.php:67]: purpose_session -> session_purpose

File: app/Http/Controllers/SessionFeedbackController.php [hunk 4]
```diff
@@ -43,12 +43,14 @@ public function store(Request $request)
             'feedback' => 'required|min:10|max:100',
             'next_session_date' => 'required',
             'next_session_time' => 'required',
+            'student_feedback' => 'required',
```
Comment (ithqindia) [inline app/Http/Controllers/SessionFeedbackController.php:46]: student_feedback -> feedback_for_student

File: app/Http/Controllers/SessionFeedbackController.php [hunk 5]
```diff
@@ -43,12 +43,14 @@ public function store(Request $request)
...
         ]);
         SessionFeedback::create([
             'session_purpose' => $request->get('session_purpose'),
             'feedback' => $request->get('feedback'),
             'next_session_date' => $request->get('next_session_date'),
             'next_session_time' => $request->get('next_session_time'),
+            'student_feedback' => $request->get('student_feedback'),
```
Comment (ithqindia) [inline app/Http/Controllers/SessionFeedbackController.php:53]: student_feedback -> feedback_for_student

File: app/Http/Controllers/SessionFeedbackController.php [hunk 6]
```diff
@@ -98,6 +101,7 @@ public function update(Request $request, $id)
         $sessionFeedback->feedback = $request->get('feedback');
         $sessionFeedback->next_session_date = $request->get('next_session_date');
         $sessionFeedback->next_session_time = $request->get('next_session_time');
+        $sessionFeedback->student_feedback = $request->get('student_feedback');
```
Comment (ithqindia) [inline app/Http/Controllers/SessionFeedbackController.php:104]: student_feedback -> feedback_for_student

File: database/migrations/2022_01_17_085955_add_fields_to_session.php [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('counselor_sessions', function (Blueprint $table) {
+            $table->string('purpose_session');
```
Comment (ithqindia) [inline database/migrations/2022_01_17_085955_add_fields_to_session.php:17]: purpose_session -> session_purpose

File: database/migrations/2022_01_17_091430_add_fields_to_student_feeedback.php [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('session_feedbacks', function (Blueprint $table) {
+            $table->string('student_feedback');
```
Comment (ithqindia) [inline database/migrations/2022_01_17_091430_add_fields_to_student_feeedback.php:17]: student_feedback -> feedback_for_student

**Review by ithqindia — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/CounselorSessionController.php (hunk 1 above)
Comment (sumedh-mohod) [reply app/Http/Controllers/CounselorSessionController.php]: done

File: app/Http/Controllers/CounselorSessionController.php (hunk 2 above)
Comment (sumedh-mohod) [reply app/Http/Controllers/CounselorSessionController.php]: done

File: app/Http/Controllers/CounselorSessionController.php (hunk 3 above)
Comment (sumedh-mohod) [reply app/Http/Controllers/CounselorSessionController.php]: done

File: app/Http/Controllers/SessionFeedbackController.php (hunk 4 above)
Comment (sumedh-mohod) [reply app/Http/Controllers/SessionFeedbackController.php]: done

File: app/Http/Controllers/SessionFeedbackController.php (hunk 5 above)
Comment (sumedh-mohod) [reply app/Http/Controllers/SessionFeedbackController.php]: done

File: app/Http/Controllers/SessionFeedbackController.php (hunk 6 above)
Comment (sumedh-mohod) [reply app/Http/Controllers/SessionFeedbackController.php]: done

File: database/migrations/2022_01_17_085955_add_fields_to_session.php (hunk 7 above)
Comment (sumedh-mohod) [reply database/migrations/2022_01_17_085955_add_fields_to_session.php]: done

File: database/migrations/2022_01_17_091430_add_fields_to_student_feeedback.php (hunk 8 above)
Comment (sumedh-mohod) [reply database/migrations/2022_01_17_091430_add_fields_to_student_feeedback.php]: done

**Comment by sumedh-mohod** [comment]
counselor form saperated and also added in frontend

File: app/Http/Controllers/CounselorSessionController.php [hunk 9]
```diff
@@ -5,65 +5,114 @@
...
+        // $user = User::find($id);
+        // if ($counselorSessionInformation) {
+        //     // If data is present then show data
+        //     return view('admin.counselor-session', compact('user', 'counselorSessionInformation'));
+        // } else {
+        //     // If no data is present then show form
+        //     $actionUrl = "/admin/students/$id/counselor-session";
+        //     return view('admin.counselor-session-form', compact('user', 'actionUrl'));
+        // }
+
```
Comment (ithqindia) [inline app/Http/Controllers/CounselorSessionController.php:68]: Remove commented code

File: app/Http/Controllers/CounselorSessionController.php [hunk 10]
```diff
@@ -5,65 +5,114 @@
...
-        $actionUrl = "/admin/students/$id/counselor-session";
-        $counselorSessionInformation = CounselorSession::where('user_id', $id)->get()->first();
-        $user = User::find($id);
-        return view('admin.counselor-session-form', compact('user', 'counselorSessionInformation', 'actionUrl'));
+        // $actionUrl = "/admin/students/$id/counselor-session";
+        // $counselorSessionInformation = CounselorSession::where('user_id', $id)->get()->first();
+        // $user = User::find($id);
+        // return view('admin.counselor-session-form', compact('user', 'counselorSessionInformation', 'actionUrl'));
+
+
```
Comment (ithqindia) [inline app/Http/Controllers/CounselorSessionController.php:81]: Remove

File: app/Http/Controllers/StudentController.php [hunk 11]
```diff
@@ -36,11 +36,13 @@ public function index()
...
     {
         $student = User::find($id);
-        $counselorSessionInformation = CounselorSession::where('user_id', $id)->get()->first();
+        $counselorSessionInformation = CounselorSession::where('user_id', $id)->get();
         $selfieInformation = Selfie::where('user_id', $id)->get()->first();
 
         $personalInformation = PersonalInformation::where('user_id', $id)->get()->first();
 
+        // $employment = EmploymentDetail::where('user_id', $id)->get();
+
```
Comment (ithqindia) [inline app/Http/Controllers/StudentController.php:45]: Remove

**Review by ithqindia — CHANGES_REQUESTED** [review]

File: resources/views/me/counselor-session-form.blade.php [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+                    </form>
+                </div>
+            </section>
+        </div>
+    </section>
+     <!-- Option 1: Bootstrap Bundle with Popper -->
+     <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/js/bootstrap.bundle.min.js"
+     integrity="sha384-ka7Sk0Gln4gmtz2MlQnikT1wXgYsOg+OMhuP+IlRH9sENBO0LRn5q+8nbTov4+1p" crossorigin="anonymous">
+ </script>
+ <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
```
Comment (ithqindia) [inline resources/views/me/counselor-session-form.blade.php:29]: Not required. It's already included in theme

File: resources/views/me/counselor-session-form.blade.php [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+     $("#purpose_input").hide();
+     $("#purpose_session").on('change', function() {
+         if ($("#purpose_session").val() == "others") {
+             $("#purpose_input").show();
+             console.log($("#purpose_session").val());
+         } else {
+             $("#purpose_input").hide();
+         }
+     });
+ </script>
```
Comment (ithqindia) [inline resources/views/me/counselor-session-form.blade.php:52]: move this in scripts section

**Review by ithqindia — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/CounselorSessionController.php (hunk 9 above)
Comment (sumedh-mohod) [reply app/Http/Controllers/CounselorSessionController.php:68]: removed

File: app/Http/Controllers/StudentController.php (hunk 11 above)
Comment (sumedh-mohod) [reply app/Http/Controllers/StudentController.php:45]: removed

File: resources/views/me/counselor-session-form.blade.php (hunk 12 above)
Comment (sumedh-mohod) [reply resources/views/me/counselor-session-form.blade.php:29]: removed

File: resources/views/me/counselor-session-form.blade.php (hunk 13 above)
Comment (sumedh-mohod) [reply resources/views/me/counselor-session-form.blade.php:52]: moved in script section

**Review by ithqindia — APPROVED** [review]

**Comment by sumedh-mohod** [comment]
view page done but showing page not found while clicked on view button

**Review by ithqindia — APPROVED** [review]

**Comment by ithqindia** [comment]
Merged in https://github.com/ithqindia/metroclub/pull/123

**Pull request closed without merging**